### PR TITLE
Fix #138 - Use cached blob urls only and make it synchronous

### LINKED ISSUE #138: Prefer blob URLs from cache whenever possible
This bug has two parts:
* Currently, in our html rewriter code [here](https://github.com/humphd/brackets/blob/bramble/src/filesystem/impls/lib/HTMLRewriter.js#L46-L52) and [here](https://github.com/humphd/brackets/blob/bramble/src/filesystem/impls/lib/HTMLRewriter.js#L83-L98), we fetch from disk and then generate a blob url. Instead, we should try to check if one already exists for that path in cache and use it and if not, only then generate a new one.
* In our css rewriter code, for now whenever we encounter a `url("path to file")` we replace it with the contents of that file. Instead, we should check for a blob url for that path in cache and use that instead, otherwise, we should generate a new one and use it instead of copy/pasting the entire contents of that file. 

cc @humphd

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

**Comment by gideonthomas** [comment]
Depends on https://github.com/humphd/brackets-browser-livedev/pull/57

File: src/filesystem/impls/filer/lib/handlers.js [hunk 1]
```diff
@@ -4,72 +4,26 @@ define(function (require, exports, module) {
...
 
-            if(Content.isHTML(ext)) {
-                _handleHTML(path, data, mimeType, callback);
-            } else if(Content.isCSS(ext)) {
-                _handleCSS(path, data, mimeType, callback);
-            } else {
-                callback(null, Content.toURL(data, mimeType));
-            }
-        });
+        return Content.toURL(data, mimeType);
```
Comment (humphd) [inline src/filesystem/impls/filer/lib/handlers.js]: In your follow up patch, we should just cache this path/data in blob utils so it gets auto-revoked on follow-up generations (e.g., reload).

**Comment by humphd** [comment]
This is great, r+

File: src/filesystem/impls/filer/lib/handlers.js (hunk 1 above)
Comment (K-LV) [inline src/filesystem/impls/filer/lib/handlers.js]: See issue: humphd/brackets#162

**Pull request merged**